src/Evolution/Systems/CurvedScalarWave/Constraints.cpp

```diff
@@ -0,0 +1,91 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/CurvedScalarWave/Constraints.hpp"
+
+#include <cstddef>
+
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+
+namespace CurvedScalarWave {
+template <size_t Dim>
+tnsr::i<DataVector, Dim, Frame::Inertial> one_index_constraint(
+    const tnsr::i<DataVector, Dim, Frame::Inertial>& d_psi,
+    const tnsr::i<DataVector, Dim, Frame::Inertial>& phi) noexcept {
+  auto constraint =
+      make_with_value<tnsr::i<DataVector, Dim, Frame::Inertial>>(phi, 0.);
+  one_index_constraint<Dim>(make_not_null(&constraint), d_psi, phi);
+  return constraint;
+}
+
+template <size_t Dim>
+void one_index_constraint(
+    const gsl::not_null<tnsr::i<DataVector, Dim, Frame::Inertial>*> constraint,
+    const tnsr::i<DataVector, Dim, Frame::Inertial>& d_psi,
+    const tnsr::i<DataVector, Dim, Frame::Inertial>& phi) noexcept {
+  if (get_size(get<0>(*constraint)) != get_size(get<0>(phi))) {
+    *constraint =
+        make_with_value<tnsr::i<DataVector, Dim, Frame::Inertial>>(phi, 0.);
+  }
+  // Declare iterators for d_psi and phi outside the for loop,
+  // because they are const but constraint is not
+  auto d_psi_it = d_psi.begin(), phi_it = phi.begin();
```

Comment:
Use `cbegin()` here to get `const_iterators`

Reply:
changed